pkg/services/accesscontrol/acimpl/accesscontrol.go

```diff
@@ -28,29 +28,36 @@ type AccessControl struct {
 	resolvers accesscontrol.Resolvers
 }
 
-func (a *AccessControl) Evaluate(ctx context.Context, user *user.SignedInUser, evaluator accesscontrol.Evaluator) (bool, error) {
+func (a *AccessControl) Evaluate(ctx context.Context, user identity.Requester, evaluator accesscontrol.Evaluator) (bool, error) {
 	timer := prometheus.NewTimer(metrics.MAccessEvaluationsSummary)
 	defer timer.ObserveDuration()
 	metrics.MAccessEvaluationCount.Inc()
 
-	if !verifyPermissions(user) {
-		a.log.Warn("no permissions set for user", "userID", user.UserID, "orgID", user.OrgID, "login", user.Login)
+	if user == nil || user.IsNil() {
```

Comment:
question: what's the difference between `user == nil` and `user.IsNil()`? They look the same to me 🤔

Reply:
the interface can be nil but be passed a nil struct inside. While `Query` structs carry a pointer to the struct and not the interface this can happen.

So it can happen that `user != nil` but `user.(*SignedInUser) == nil`

https://go.dev/play/p/Isoo0CcAvr nice playground example